app/domain/authentication/authn_oidc/v2/client.rb

```diff
@@ -97,6 +97,46 @@ def callback(code:, nonce:, code_verifier: nil)
           decoded_id_token
         end
 
+
+        # callback_with_temporary_cert wraps the callback method with commands
+        # to write & clean up a certificate to & from Conjur's default
+        # certificate store.
+        #
+        # The temporary certificate file name is "x.0", where x is the hash of
+        # the certificate subject name. If this file already exists in the
+        # default cert store, the original certificate is used.
+        #
+        # Unlike self.discover, which wraps a single ::OpenIDConnect method,
+        # callback_with_temporary_cert wraps the entire callback method, which
+        # includes multiple calls to the OIDC provider, including at least one
+        # discover! call. The temporary certs will apply to all required
+        # operations.
+        def callback_with_temporary_cert(
+          code:,
+          nonce:,
+          code_verifier: nil,
+          cert_dir: OpenSSL::X509::DEFAULT_CERT_DIR,
+          cert_string: nil
+        )
+          c = -> { callback(code: code, nonce: nonce, code_verifier: code_verifier) }
+
+          return c.call if cert_string.blank?
+
+          cert = OpenSSL::X509::Certificate.new(cert_string)
```

Comment:
Any idea if this constructor will work with a cert chain?

If not, ::Conjur::CertUtils has a `parse_certs` method which should return an array of Certificate objects

Reply:
This constructor does not work with a chain - updated to use `CertUtils.parse_certs`